Jenkinsfile

```diff
@@ -41,10 +30,26 @@ node('vagrant') {
 
         EcoSystem ecoSystem = new EcoSystem(this, "gcloud-ces-operations-internal-packer", "jenkins-gcloud-ces-operations-internal")
 
-        try {
+        stage('Checkout') {
+            checkout scm
+        }
+
+        stage('Lint') {
+            lintDockerfile()
+            shellCheck("resources/startup.sh resources/upgrade-notification.sh resources/pre-upgrade.sh")
 
+            if (env.CHANGE_TARGET) {
+                echo 'This is a pull request; checking changelog...'
+                String newChanges = changelog.changesForVersion('Unreleased')
```

Comment:
If I could make an educated guess I would say the `unstable()` method **will not be called** if 
1. a first PR adds some functionality along _with a change entry to the CHANGELOG_
   - the build might be stable because of the new change log 
2. no release happens
2. any following PRs make changes _but do not change the CHANGELOG_ any further
   - the build might be stable because of the change log from the first PR

Reply:
That being said though, I am convinced that there might be changes (like fixing docs or fixing auxiliary build files like Jenkinsfile etc) that are necessary and yet do not contain enough substance for a change log.

Reply:
@robertauer confirms my suspicion. Anyhow this solution is step forward in the direction of automatic PR findings